src/test/java/com/meilisearch/integration/SearchTest.java

```diff
@@ -217,13 +183,10 @@ public void testSearchHighlight() throws Exception {
 
 		SearchRequest searchRequest = new SearchRequest("and")
 			.setAttributesToHighlight(new String[]{"overview"});
-		Results res_gson = jsonGson.decode(
-			index.search(searchRequest),
-			Results.class
-		);
-		assertEquals(20, res_gson.hits.length);
-		assertTrue(res_gson.hits[0].getFormatted().getOverview().contains("<em>"));
-		assertTrue(res_gson.hits[0].getFormatted().getOverview().contains("</em>"));
+
+		SearchResult searchResult = index.search(searchRequest);
+
+		assertEquals(20, searchResult.getHits().size());
```

Comment:
@eskombro it can be, but seems `Movie` class is only for test, and `_formated` is the parameter only for this use case. 

As I commented above, that is why I'm thinking of using `Array<Object>` instead of `Array<HashMap>` to allow user to setup serialization...but not sure which is the better way.